extmod/nimble/modbluetooth_nimble.c

```diff
@@ -351,7 +352,11 @@ STATIC void sync_cb(void) {
     }
 
     DEBUG_printf("sync_cb: Setting device name\n");
-    ble_svc_gap_device_name_set(MICROPY_PY_BLUETOOTH_DEFAULT_GAP_NAME);
+    if (nimble_gap_name[0] != '\0') {
+        ble_svc_gap_device_name_set(nimble_gap_name);
+    } else {
+        ble_svc_gap_device_name_set(MICROPY_PY_BLUETOOTH_DEFAULT_GAP_NAME);
+    }
```

Comment:
Would it be better to just not call `ble_svc_gap_device_name_set()` if the name was already set?  I guess this depends on the internals of NimBLE, if it retains the GAP name if it was set before bringing up the stack.

Reply:
Forced-pushed an alternate commit implementing this approach.

Reply:
This looks good, thanks!